lms/product/product.py

```diff
@@ -66,6 +70,17 @@ class Product:
     # Accessor for external consumption
     Family = Family
 
+    def is_gradable(self, lti_params):
```

Comment:
Adding three new methods here.

I'm adding them at the top level class as there's not much else to generalize a new smaller one. It should be easy to clean if the interface grows to much.


Adding:

- `is_gradable`

Used in: 
https://github.com/hypothesis/lms/blob/fe9bac5786a15f8eb937bba296343526c7ba1117/lms/views/lti/basic_launch.py#L156

  Special D2L case in: 
  https://github.com/hypothesis/lms/blob/fe9bac5786a15f8eb937bba296343526c7ba1117/lms/product/d2l/product.py#L26


I reckon this ones is fine. Maybe in other LMS we'll need to take request instead or the params? We'll see when we get there.

- `ltia_aud_claim`


D2L case

https://github.com/hypothesis/lms/blob/fe9bac5786a15f8eb937bba296343526c7ba1117/lms/product/d2l/product.py#L35

Used in https://github.com/hypothesis/lms/blob/fe9bac5786a15f8eb937bba296343526c7ba1117/lms/services/ltia_http.py#L72


We might find out that we need more changes for other LMS. Again we can change it later and I like how you can go to the LTIA service see the common parameters there except for the special one, which is clearly marked.


- `configure_assignment`

Used in https://github.com/hypothesis/lms/blob/fe9bac5786a15f8eb937bba296343526c7ba1117/lms/views/lti/basic_launch.py#L132


D2L change https://github.com/hypothesis/lms/blob/fe9bac5786a15f8eb937bba296343526c7ba1117/lms/product/d2l/product.py#L41 

(which the main focus of this PR)

This one might be a bit to broad. It's a big hook to do "whatever" while configuring the assignment.

What we are doing here is at the moment something that only makes sense in D2L (ie, not something we are deciding not to to elsewhere, but something that doesn't make sense to do). 

We could, in the the main `configure_assignment` in the view use a setting to conditionally do it there, based on the setting, not the product.

I reckon in that scenario we'd need a comment explaining the situation in D2L so that suggests it belongs to the product.